grafana-plugin/src/containers/OutgoingWebhookForm/OutgoingWebhookForm.tsx

```diff
@@ -59,165 +50,139 @@ export const WebhookTabs = {
+  const data = { ...rawData };
+  selectedPreset.controlled_fields.forEach((field) => {
+    delete data[field];
+  });
+
+  return data;
+}
 
 export const OutgoingWebhookForm = observer((props: OutgoingWebhookFormProps) => {
-  const history = useHistory();
   const { id, action, onUpdate, onHide, onDelete } = props;
-  const [onFormChangeFn, setOnFormChangeFn] = useState<{ fn: (value: string) => void }>(undefined);
-  const [templateToEdit, setTemplateToEdit] = useState(undefined);
-  const [activeTab, setActiveTab] = useState<string>(
-    action === WebhookFormActionType.EDIT_SETTINGS ? WebhookTabs.Settings.key : WebhookTabs.LastRun.key
-  );
-  const [showPresetsListDrawer, setShowPresetsListDrawer] = useState(id === 'new');
-  const [showCreateWebhookDrawer, setShowCreateWebhookDrawer] = useState(false);
+
   const [selectedPreset, setSelectedPreset] = useState<OutgoingWebhookPreset>(undefined);
-  const [filterValue, setFilterValue] = useState('');
+  const [templateToEdit, setTemplateToEdit] = useState<TemplateParams>(undefined);
+
+  const { outgoingWebhookStore } = useStore();
 
-  const { outgoingWebhookStore, hasFeature, grafanaTeamStore, alertReceiveChannelStore } = useStore();
-  const isNew = action === WebhookFormActionType.NEW;
-  const isNewOrCopy = isNew || action === WebhookFormActionType.COPY;
-  const form = createForm({
-    presets: outgoingWebhookStore.outgoingWebhookPresets,
-    grafanaTeamStore,
-    alertReceiveChannelStore,
-    hasLabelsFeature: hasFeature(AppFeature.Labels),
+  const item = outgoingWebhookStore.items[id];
+  const data = prepareDataForEdit(action, item, selectedPreset);
+
+  useEffect(() => {
+    if (item) {
+      const preset = outgoingWebhookStore.outgoingWebhookPresets.find((item) => item.id === data.preset);
+      setSelectedPreset(preset);
+    }
+  }, [item]);
+
+  useEffect(() => {
+    if (selectedPreset) {
+      reset(data);
+    }
+  }, [selectedPreset]);
+
+  const formMethods = useForm<ApiSchemas['Webhook']>({
+    mode: 'onChange',
+    defaultValues: data,
   });
 
-  const handleSubmit = useCallback(
-    async (data: Partial<ApiSchemas['Webhook']>) => {
+  const { setValue, handleSubmit, reset } = formMethods;
+
+  const onSubmit = useCallback(
+    async (rawData: Partial<ApiSchemas['Webhook']>) => {
+      const data = prepareForSave(rawData, selectedPreset);
+
       try {
-        if (isNewOrCopy) {
+        if (action === WebhookFormActionType.NEW || action === WebhookFormActionType.COPY) {
           await outgoingWebhookStore.create(data);
         } else {
           await outgoingWebhookStore.update(id, data);
         }
-        outgoingWebhookStore.setLabelsFormErrors(undefined);
         onHide();
         onUpdate();
-      } catch (err) {
-        if (err.response?.data?.labels) {
-          outgoingWebhookStore.setLabelsFormErrors(err.response.data.labels);
-        }
+      } catch (error) {
+        /*  Object.keys(error.response.data).forEach((key) => {
```

Comment:
Do we need this comment? If yes, could you put one line more with explanation? If not, let's remove it

Reply:
Ive uncommented, it injects errors into form that come from BE